kafkatest/broker.go

```diff
@@ -195,7 +208,7 @@ func (p *Producer) Produce(topic string, partition int32, messages ...*proto.Mes
 		Partition: partition,
 		Messages:  messages,
 	}
-	p.ResponseOffset += int64(len(messages))
+	p.responseOffset = off
```

Comment:
I don't think this line should have changed (except to reflect `responseOffset` becoming private).  This just sets it to the same value it was before.